### PR TITLE
dependencies: Set range versions for the dependencies

### PR DESCRIPTION
**Pull request by psaiz**
Set the version for the invenio dependencies as a range
Increase the minimum dependency on invenio-search-ui
Increase version number of xrootd to be compatible with the Dockerfile

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -67,36 +67,36 @@
...
-    "invenio-records-files==1.2.1",
     # Custom Invenio `postgresql` bundle
-    "invenio-db[versioning,postgresql]==1.1.0",
+    "invenio-db[versioning,postgresql]>=1.1.0,<2",
     # Custom Invenio `opensearch` bundle
-    "invenio-search[opensearch2]==2.1.0",
+    "invenio-search[opensearch2]>=2.1.0,<3",
     # Specific Invenio dependencies
-    "invenio-xrootd>=1.0.0a6",
+    "invenio-xrootd>=1.0.0a6,<2",
```
Comment (zzacharo) [inline setup.py:99]: Maybe because this is an alpha I would be stricter and pin it....wdyt?

File: setup.py [hunk 2]
```diff
@@ -67,36 +67,36 @@
 
 install_requires = [
     # General Invenio dependencies
-    "invenio-app==1.3.0",
-    "invenio-base==1.3.0",
-    "invenio-config==1.0.3",
+    "invenio-app>=1.3.0,<2",
```
Comment (zzacharo) [inline setup.py:70]: cosmetic: 
```suggestion
    "invenio-app>=1.3.0,<2.0.0",
```

File: setup.py (hunk 1 above)
Comment (psaiz) [reply setup.py]: Yes, sure.

File: setup.py (hunk 2 above)
Comment (psaiz) [reply setup.py]: Ok, changed all of them

File: setup.py [hunk 3]
```diff
@@ -67,36 +67,36 @@
...
+    "invenio-config>=1.0.3,<2.0.0",
     # Custom Invenio `base` bundle
-    "invenio-assets==3.0.0",
-    "invenio-accounts==3.0.0",
     "importlib-metadata==4.13.0",
-    "invenio-logging[sentry]==1.3.0",
+    "invenio-accounts>=3.0.0,<4.0.0",
+    "invenio-assets>=3.0.0,<4.0.0",
+    "invenio-logging[sentry]>=2.1.0,<3.0.0",
     "invenio-rest==1.2.8",
```
Comment (psaiz) [inline setup.py:78]: This one was not changed. The latest version requires a new mayor of invenio-logging, and the sentry dependencies are different

File: setup.py [hunk 4]
```diff
@@ -67,36 +67,36 @@
...
+    "invenio-search-ui>=2.8.3,<3.0.0",
     # Custom Invenio `files` bundle
-    "invenio-previewer==2.0.1",
+    "invenio-previewer>=2.0.1,<3.0.0",
+    "invenio-records-files>=1.2.1,<3.0.0",
     "jupyter-client==7.1.0",
     "pluggy==0.13.1",
-    "invenio-records-files==1.2.1",
     # Custom Invenio `postgresql` bundle
     "invenio-db[versioning,postgresql]==1.1.0",
```
Comment (psaiz) [inline setup.py:95]: This one was not changed. The newer versions require a new mayor of Flask-SQLAlchemy, and I got scared by that one

**Review by zzacharo — APPROVED** [review]

**Pull request merged**